### PR TITLE
Add exponential backoff to elastic.bulk

### PR DESCRIPTION
**Pull request by ZJONSSON**

### CODE REVIEW

File: lib/elasticsearch/bulk.js [hunk 1]
```diff
@@ -108,13 +110,33 @@ Bulk.prototype._fn = function(d) {
       });
       return items.length && items || undefined;
     },e => {
-      retry = (retry || 0);
-      if (!this.options.maxRetries || retry > this.options.maxRetries)
-        throw e;
+      retries = retries || [];
+      const retryNo = retries.length;
+      if (!this.options.maxRetries || retryNo >= this.options.maxRetries) {
+        if (e) e.retries = retries;
```
Comment (jbreckman) [inline lib/elasticsearch/bulk.js]: is this just for visibility?

File: test/elastic-retry-test.js [hunk 2]
```diff
@@ -30,5 +39,44 @@ t.test('elastic bulk insert',async t => {
...
+  const upsert = etl.elastic.upsert(backoffFailingClient,'test','test', options);
+  let err;
+
+  await etl.toStream([1,2,3].map(function(d) { return {_id:d,num:d};}))
+    .pipe(etl.collect(3))
+    .pipe(upsert)
+    .promise()
+    .catch(e => err = e);
+
+  t.same(err.retries, [10, 20, 40, 80, 160, 250, 250], 'exponential backoff with a cap');
```
Comment (jbreckman) [inline test/elastic-retry-test.js]: oh i see

File: lib/elasticsearch/bulk.js (hunk 1 above)
Comment (ZJONSSON) [reply lib/elasticsearch/bulk.js]: Mostly for testing, i.e. showing the retries on the error object allows us to inspect.

**Comment by ZJONSSON** [comment]
Thanks @jbreckman !!

**Comment by ZJONSSON** [comment]
published as `+ etl@0.5.22`

**Pull request merged**